### PR TITLE
fix test-case vs devise vs rails 4.2

### PR DESCRIPTION
**Pull request by grosser**
should solve #23 and #32 

@rafaelfranca @walterdavis @guigs

### CODE REVIEW

**Comment by grosser** [comment]
@rafaelfranca @walterdavis @guigs

**Comment by guigs** [comment]
I've applied this to 4.2 branch in my fork of prototype-rails and it is working with my app.

File: lib/prototype-rails/on_load_action_view.rb [hunk 1]
```diff
@@ -14,9 +14,11 @@
   include ActionView::Helpers::ScriptaculousHelper
 end
 
-ActionView::TestCase.class_eval do
-  include ActionView::Helpers::PrototypeHelper
-  include ActionView::Helpers::ScriptaculousHelper
+if defined?(Rails) && Rails.env.test?
```
Comment (rafaelfranca) [inline lib/prototype-rails/on_load_action_view.rb]: Can we check `defined?(Rails.env)`?

Comment (rafaelfranca) [inline lib/prototype-rails/on_load_action_view.rb]: The problem with this is if users is testing in a cucumber environment it will not load these helpers.

Comment (grosser) [inline lib/prototype-rails/on_load_action_view.rb]: the idea was to only load it in test mode, so `defined?(Rails.env)` will also be true for development and others

how about `['cucumber', 'test'].include?(Rails.env)` ?
there is always the fallback option of doing the include manually, so even if we miss some edge case it's not that bad

Comment (rafaelfranca) [inline lib/prototype-rails/on_load_action_view.rb]: I mean `if defined?(Rails.env) && Rails.env.test?`.

I really thing we should not check any environment but I can't think in a better option besides a configuration.

Comment (grosser) [inline lib/prototype-rails/on_load_action_view.rb]: Ah yes, that should work and is a bit safer, changed!

**Comment by grosser** [comment]
@rafaelfranca good to go now ?

**Comment by rafaelfranca** [comment]
I still don't think checking environment is a good option :cry:, but I guess there is no other way of doing it.

**Comment by grosser** [comment]
Let's wait for the next PR complaining on this ;)
